demo/midgard-sdk/src/tx-builder/user-events/deposit/index.ts

```diff
@@ -1 +1,96 @@
+import {
+  Data,
+  Integer,
+  LucidEvolution,
+  RedeemerBuilder,
+  Script,
+  toUnit,
+  TransactionError,
+  TxBuilder,
+} from "@lucid-evolution/lucid";
+import { Datum, DepositInfo, MintRedeemer } from "./types.js";
+import { POSIXTime } from "@/tx-builder/common.js";
+import { bufferToHex, hashHexWithBlake2b256 } from "@/utils/common.js";
+import { Effect } from "effect";
+
+export type DepositParams = {
+  depositScriptAddress: string;
+  mintingPolicy: Script;
+  policyId: string;
+  depositInfo: DepositInfo;
+  inclusionTime: POSIXTime;
+};
+
+/**
+ * Deposit
+ *
+ * @param lucid - The LucidEvolution
+ * @param params - The parameters
+ * @returns {TxBuilder} A TxBuilder instance that can be used to build the transaction.
+ */
+
+export const depositTxBuilder = (
+  lucid: LucidEvolution,
+  params: DepositParams,
+): Effect.Effect<TxBuilder, TransactionError | unknown> =>
+  Effect.gen(function* () {
+    const utxos = yield* Effect.promise(() => lucid.wallet().getUtxos());
+    if (utxos.length === 0) {
+      throw new Error("No UTxOs found in wallet");
+    }
+    const inputUtxo = utxos[0];
+    const assetNameBuffer = yield* hashHexWithBlake2b256(inputUtxo.txHash);
+    const assetName = bufferToHex(Buffer.from(assetNameBuffer, "hex"));
+    const depositNFT = toUnit(params.policyId, assetName);
+    const currDatum: Datum = {
+      event: {
+        id: {
+          txHash: { hash: inputUtxo.txHash },
+          outputIndex: BigInt(inputUtxo.outputIndex),
+        },
+        info: params.depositInfo,
+      },
+      inclusionTime: params.inclusionTime,
+    };
+    const depositDatum = Data.to(currDatum, Datum);
+    const authenticateEvent: RedeemerBuilder = {
+      kind: "selected",
+      makeRedeemer: (inputIdxs: bigint[]) => {
+        const tupleList = inputIdxs.map((inputIdx, i) => [inputIdx, BigInt(i)]);
+
+        const redeemer: MintRedeemer = {
+          AuthenticateEvent: {
+            nonceInputIndex: tupleList[0][0],
+            eventOutputIndex: 0n,
+            hubRefInputIndex: 0n,
+            witnessRegistrationRedeemerIndex: 0n,
+          },
+        };
+
+        return Data.to(redeemer, MintRedeemer);
+      },
+      inputs: [inputUtxo],
+    };
```

Comment:
Thanks for the attempt, but as mentioned in DMs, LE's interface probably won't cut it for our usecase here. Let's continue without the `RedeemerBuilder` for now.